df_engine_extensions/src/dist_sql_query/resolver.rs

```diff
@@ -115,9 +117,14 @@ impl Resolver {
     }
 
     pub fn resolve_partitioned_scan_internal(
-        &self,
+        &mut self,
         plan: Arc<dyn ExecutionPlan>,
     ) -> DfResult<Arc<dyn ExecutionPlan>> {
+        // Check if this plan is `AnalyzeExec`, if it is, we should collect metrics.
```

Comment:
It is good to do so, but maybe we just pass the `is_analyze` into `ResolvedPartitionedScan`, and don't represent it by setting `metrics_collector`, due to I want to remove `metrics_collector` after...